homeassistant/components/vartastorage/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The VARTA Storage integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from vartastorage.vartastorage import VartaStorage
+
+from homeassistant import config_entries, core
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+PLATFORMS: list[str] = ["sensor"]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up VARTA Storage from a config entry."""
+    varta = VartaStorage(entry.data["host"], entry.data["port"])
+    try:
+        varta.client.connect()
+    except Exception as ex:
+        _LOGGER.error("Could not connect to modbus server")
```

Comment:
On connection error, you should [log a warning rather than an error](https://developers.home-assistant.io/docs/integration_quality_scale_index#silver-).

Reply:
done